### PR TITLE
GaugeController: Add Owner and Manager Roles

### PR DESCRIPTION
**Pull request by kamescg**

### CODE REVIEW

File: test/GaugeController.test.ts [hunk 1]
```diff
@@ -176,6 +177,11 @@ describe('GaugeController', () => {
                 '1000000000000000000',
             );
         });
+
+        it('should FAIL to execute BECAUSE of unauthorized access', async () => {
```
Comment (PierrickGT) [inline test/GaugeController.test.ts:181]: We have no tests for `addGauge`?

File: test/GaugeController.test.ts [hunk 2]
```diff
@@ -220,8 +231,29 @@ describe('GaugeController', () => {
                 '500000000000000000',
             );
         });
+
+        it('should FAIL to execute BECAUSE of unauthorized access', async () => {
```
Comment (PierrickGT) [inline test/GaugeController.test.ts:235]: We should also test that the function can be executed by a manager.

File: test/GaugeController.test.ts [hunk 3]
```diff
@@ -220,8 +231,29 @@ describe('GaugeController', () => {
...
+            await GaugeController.addGauge(gaugeAddress);
+            const unauthorized = GaugeController.connect(wallet2);
+            expect(unauthorized.setGaugeScale(gaugeAddress, toWei('2'))).to.be.revertedWith('Manageable/caller-not-manager-or-owner');
+        });
     });
 
+     describe('setGaugeReward(IGaugeReward _gaugeReward)', () => {
+        it('should SUCCEED to SET a new GaugeReward address', async () => {
```
Comment (PierrickGT) [inline test/GaugeController.test.ts:243]: We should also test that the function can be executed by a manager.

**Review by PierrickGT — CHANGES_REQUESTED** [review]
Looks good to me. Just need to add a couple tests for the manager role and also remove the `/// @TODO: Add Governance/Executive authorization modifier/function.` comments.

File: contracts/GaugeController.sol [hunk 4]
```diff
@@ -238,7 +238,7 @@ contract GaugeController is IGaugeController, Ownable {
      * @notice Add new gauge with "1e18" scale to the GaugeController.
      * @param _gauge Address of the Gauge
      */
-    function addGauge(address _gauge) public {
+    function addGauge(address _gauge) public onlyOwner {
```
Comment (asselstine) [inline contracts/GaugeController.sol:241]: external

File: contracts/GaugeController.sol [hunk 5]
```diff
@@ -248,7 +248,7 @@ contract GaugeController is IGaugeController, Ownable {
      * @param _gauge Address of new Gauge
      * @param _scale Amount to scale new Gauge by
     */
-    function addGaugeWithScale(address _gauge, uint256 _scale) public {
+    function addGaugeWithScale(address _gauge, uint256 _scale) public onlyOwner {
```
Comment (asselstine) [inline contracts/GaugeController.sol:251]: external

File: contracts/GaugeController.sol [hunk 6]
```diff
@@ -257,7 +257,7 @@ contract GaugeController is IGaugeController, Ownable {
      * @notice Remove gauge from the GaugeController.
      * @param _gauge Address of existing Gauge
     */
-    function removeGauge(address _gauge) public {
+    function removeGauge(address _gauge) public onlyOwner {
```
Comment (asselstine) [inline contracts/GaugeController.sol:260]: external

**Pull request merged**